application-diagram-api/src/main/java/com/xwiki/diagram/internal/handlers/DiagramMacroSolrMetadataExtractor.java

```diff
@@ -65,32 +73,81 @@ public class DiagramMacroSolrMetadataExtractor implements SolrEntityMetadataExtr
     @Inject
     private EntityNameValidationConfiguration entityNameValidationConfiguration;
 
+    @Inject
+    private Logger logger;
+
     @Override
     public boolean extract(XWikiDocument document, SolrInputDocument solrDocument)
     {
 
-        List<Block> macroBlocks = document.getXDOM().getBlocks(new MacroBlockMatcher("diagram"), Block.Axes.CHILD);
-        if (macroBlocks != null && macroBlocks.size() > 0) {
+        XDOM xdom = document.getXDOM();
+        List<Block> macroBlocks = xdom.getBlocks(new MacroBlockMatcher("diagram"), Block.Axes.CHILD);
+        if (macroBlocks != null && macroBlocks.size() > 0 && !updateMacroReference(document, xdom, macroBlocks)) {
             List<DocumentReference> macroReferences = new ArrayList<>();
             for (Block macroBlock : macroBlocks) {
-                String referenceName =
-                    macroBlock.getParameter(REFERENCE) != null ? macroBlock.getParameter(REFERENCE) : "Diagram";
-
-                String transformedNamed = this.transformName(referenceName);
 
                 DocumentReference macroReference =
-                    explicitDocumentReferenceResolver.resolve(transformedNamed, document.getDocumentReference());
+                    explicitDocumentReferenceResolver.resolve(macroBlock.getParameter(REFERENCE),
+                        document.getDocumentReference());
                 macroReferences.add(macroReference);
             }
             return linkRegistry.registerBacklinks(solrDocument, macroReferences);
+        }
+        return false;
+    }
 
+    /**
+     * Checks and updates the references of all the diagram macro calls to make sure that all of them respect the
+     * current name strategy.
+     *
+     * @param document document with all the macro calls
+     * @param xdom of the @document
+     * @param macroBlocks list of all the diagram macro calls
+     * @return true if all the references are valid, false otherwise.
```

Comment:
` true if references are valid` sounds misleading. What I understand is that it returns true if references are not valid and they have been updated, false if there were no invalid references

Reply:
You are right the message should have been "true if any reference is invalid, false otherwise"

Reply:
I still find it incomplete. What about `true if any reference was invalid and has been updated, false if there weren't any invalid references or if an error occurred while updating the document`